### PR TITLE
processing data in models

### PR DESCRIPTION
**Pull request by mntedev**
**I have completed the following tasks.** :hand: 

:heavy_check_mark:  Added association between models
:heavy_check_mark:  Created two users via Rails console.
:heavy_check_mark:  Created five posts via Rails console
:heavy_check_mark:   Created comments and likes via console
:heavy_check_mark:  Created a method to return the 3 most recent posts for a given user.
:heavy_check_mark:  Created a method to update the posts counter for a user
:heavy_check_mark:  created a method to return the 5 most recent comments for a given post
:heavy_check_mark: Created a method to  updates the comments counter for a post

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -1,2 +1,9 @@
 class User < ApplicationRecord
+  has_many :posts
+  has_many :comments
+  has_many :likes
```
Comment (rloterh) [inline app/models/user.rb]: - [ ] As per the given _ERD_, the _User_ **association** should have _author_id_ referenced as **foreign key** in _Post_, _Comment,_ and _Like_ models. Kindly ensure to improve this aspect, as can be seen below:
```ruby
has_many :posts, foreign_key: 'author_id'
has_many :comments, foreign_key: 'author_id'
has_many :likes, foreign_key: 'author_id'
```

File: app/models/like.rb [hunk 2]
```diff
@@ -1,2 +1,8 @@
 class Like < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  belongs_to :post
+  after_save :update_likes_counter
+  def update_likes_counter
+    post.upadate(likes_counter: post.likes.count)
+  end
```
Comment (rloterh) [inline app/models/like.rb:10]: - [ ] OPTIONAL
Following best practices `update_likes_counter` should be marked as private after implementing the `after_save` method.

File: app/models/comment.rb [hunk 3]
```diff
@@ -1,2 +1,8 @@
 class Comment < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  belongs_to :post
+  after_save :update_comment_count
+  def update_comment_count
+    post.upadate(comments_counter: post.comments.count)
+  end
```
Comment (rloterh) [inline app/models/comment.rb:10]: - [ ] OPTIONAL
Following best practices `update_comment_count` should be marked as private after implementing the `after_save` method.

File: app/models/post.rb [hunk 4]
```diff
@@ -1,2 +1,13 @@
 class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  has_many :comments
+  has_many :likes
+  after_save :update_posts_counter
+  def update_posts_counter
+    author.upadate(posts_counter: author.posts.count)
+  end
```
Comment (rloterh) [inline app/models/post.rb:11]: - [ ] OPTIONAL
Following best practices `update_posts_counter` should be marked as private after implementing the `after_save` method.

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Worash, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there! 🚦 🚦 🚦  

## Highlights 👍  
- Descriptive Pull Request
- No linter errors


## Required Changes ♻️
_Kindly check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

_Kindly check the comments marked **[OPTIONAL]** under the review._

**You can also consider:**
- It's recommended that you add an MIT license, or any other similar license to your ReadMe.

<br>
Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by m05e5 — APPROVED** [review]
# Status: ✔️ Project Approved ✔️ 

Hello @worashf  🔥 

Good job on implementing the required changes. Well Done.
Your project is complete! There is nothing else to say other than... it's time to merge it :shipit:
Congratulations! 🎉

goodjob


Now you can merge this branch 🚀 

Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.


_m05e5_

### 💻 👩🏽‍💻  Happy Coding!  👨🏽‍💻 🖥️

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**